### PR TITLE
fix(flutterfire_ui): Fix incorrect state when country code is null.

### PR DESCRIPTION
**Pull request by richard457**
## Description

Currently on phone auth, when the user deletes country code it fires an error bad state, which is normal behavior of  countries.firstWhere( 

## Related Issues

Fixes *https://github.com/FirebaseExtended/flutterfire/issues/7694*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x]  I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x]  I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by russellwheatley** [comment]
Hey @richard457, do you mind signing the CLA so we can get this merged in, please? Thanks.

File: packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart [hunk 1]
```diff
@@ -135,6 +135,7 @@ class PhoneInputState extends State<PhoneInput> {
         (element) =>
             element.countryCode == countryCode ||
             element.phoneCode == phoneCode,
+        orElse: () => countries.first,
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart:138]: I'm not entirely sure what exactly this PR intends to fix. This block of code is wrapped with try/catch and the case where no country code is found is handled in catch block

**Comment by lesnitsky** [comment]
This PR doesn't fix an issue, since there is no issue in first place. The `BadState` exception is being handled properly by a catch block

File: packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart (hunk 1 above)
Comment (richard457) [reply packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart:138]: Except the catch failed to catch the error, at least on my end. and that was the bug fired.

Comment (richard457) [reply packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart:138]: @lesnitsky I am not sure why you closed this without deep investigation, for example, an assumption that a catch will catch all errors is a bad assumption, then why the bad state was firing? I am a bit confused in this case.

Comment (lesnitsky) [reply packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart:138]: @richard457 there is no such scenario where the exception wouldn't be caught in the code above

```dart
try {
      final newItem = countries.firstWhere(
        (element) =>
            element.countryCode == countryCode ||
            element.phoneCode == phoneCode,
      );
      // ...
    } catch (_) {
      // ...
    }
```

I think you had "All exceptions" checked in vscode debugger

**Pull request closed without merging**